File: Java/백준/Silver/1654. 랜선 자르기/랜선 자르기.java
import java.util.*;
import java.io.*;

public class Main{
	public static int arr[];
	public static int count=0;

	public static void main(String[] args)throws IOException {

		Scanner sc = new Scanner(System.in);
		int a=sc.nextInt();
		int b=sc.nextInt();
		long arr[]=new long[a];
		long max=0;
		
		
		for(int i=0;i<a;i++) {
			arr[i]=sc.nextInt();
			if(arr[i]>max)
				max=arr[i];
		}
		long min=1;
		long mid=0;
		
		
		
		while(min<=max) {
			mid=(max+min)/2;
			long count=0;
			for(int i=0;i<arr.length;i++) {
				count+=(arr[i]/mid);
			}
			if(count<b) {
				max=mid-1;
			}
			else {
				min=mid+1;
			}
		}
		System.out.println(min-1);
		
		
	}

}

File: Swift/백준/Silver/1654. 랜선 자르기/랜선 자르기.swift
import Foundation



let s = "sdsd"


let input = readLine()!.split(separator: " ")

let n = Int(input[0])!
let k = Int(input[1])!

var start = 1
var end = 0
var result = 0
var mid = 0

var arr = Array(repeating: 0, count: n)
for i in 0..<arr.count {
    let a = Int(readLine()!)!
    arr[i] = a
    if arr[i] >= end {
        end = arr[i]
    }
}

while start <= end {
    var sum = 0
    mid = (start+end)/2
    for i in 0..<arr.count {
        sum += (arr[i]/mid)
    }
    if sum >= k {
        start = mid + 1
  //      end = mid - 1
        result = mid
    } else {
        end = mid - 1
    }
}
print(result)
